core/runtime/src/main/java/io/quarkus/runtime/ShutdownEvent.java

```diff
@@ -14,4 +15,43 @@
  * The annotated method can access other injected beans.
  */
 public class ShutdownEvent {
+
+    private ShutdownReason shutdownReason;
```

Comment:
Speaking of nit-picking - this field should be `final` :-)

Reply:
Fixed via https://github.com/quarkusio/quarkus/pull/26911